nixos/modules/services/web-apps/keycloak.nix

```diff
@@ -0,0 +1,71 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.keycloak;
+in {
+  options = {
+    services.keycloak = {
+      enable = mkEnableOption "Keycloak identity and access server";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.keycloak;
+        defaultText = "pkgs.keycloak";
+        example = literalExample "pkgs.keycloak";
+        description = "Package that should be used for keycloak";
+      };
+
+      jbossBaseDir = mkOption {
+        type = types.str;
+        default = "/var/lib/keycloak";
+        description = ''
+          The location keycloak (and underlying JBoss/Wildfly) information
+          (configuration, logs, data) is stored.
+        '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        description = "The user to run keycloak as.";
+        default = "keycloak";
+      };
+
+      group = mkOption {
+        type = types.str;
+        description = "The group to run keycloak as.";
+        default = "keycloak";
+      };
+
+      extraFlags = mkOption {
+        description = "Extra flags to pass to the keycloak (standalone.sh) command.";
+        default = "";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.keycloak-server = {
+      after = [ "network.target" ];
+      description = "Keycloak Open Source Identity and Access Management Server";
+
+      environment = {
+        JBOSS_BASE_DIR= jbossBaseDir;
+      };
+
+      serviceConfig = {
+        User = "${cfg.user}";
+        Group = "${cfg.group}";
+        TimeoutStartSec=60;
```

Comment:
- please be consistent with spacing
- is `60` sufficient? I ask because I see `600` here: https://www.techrunnr.com/how-to-set-up-a-keycloak-server-in-linux/ and we're dealing with `java`, which can take a while...

Reply:
I've applied https://github.com/nix-community/nixpkgs-fmt, hope that helps with the spacing?

I'm hesitant to have more than 60s- I'd like to make sure it's faster than that, even with java, as I get pretty annoyed by some of the other slow to start/terminate services I end up having around.